third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -0,0 +1,405 @@
+            Operation::Gt => self.gen_op_call(targets, id, BytecodeOperation::Gt, args),
+            Operation::Le => self.gen_op_call(targets, id, BytecodeOperation::Le, args),
+            Operation::Ge => self.gen_op_call(targets, id, BytecodeOperation::Ge, args),
+            Operation::Not => self.gen_op_call(targets, id, BytecodeOperation::Not, args),
+
+            // Non-supported specification related operations
+            Operation::SpecFunction(_, _, _)
+            | Operation::Implies
+            | Operation::Iff
+            | Operation::Tuple
+            | Operation::Select(_, _, _)
+            | Operation::UpdateField(_, _, _)
+            | Operation::Index
+            | Operation::Slice
+            | Operation::Range
+            | Operation::Result(_)
+            | Operation::Len
+            | Operation::TypeValue
+            | Operation::TypeDomain
+            | Operation::ResourceDomain
+            | Operation::Global(_)
+            | Operation::CanModify
+            | Operation::Old
+            | Operation::Trace(_)
+            | Operation::Identical
+            | Operation::EmptyVec
+            | Operation::SingleVec
+            | Operation::UpdateVec
+            | Operation::ConcatVec
+            | Operation::IndexOfVec
+            | Operation::ContainsVec
+            | Operation::InRangeRange
+            | Operation::InRangeVec
+            | Operation::RangeVec
+            | Operation::MaxU8
+            | Operation::MaxU16
+            | Operation::MaxU32
+            | Operation::MaxU64
+            | Operation::MaxU128
+            | Operation::MaxU256
+            | Operation::Bv2Int
+            | Operation::Int2Bv
+            | Operation::AbortFlag
+            | Operation::AbortCode
+            | Operation::WellFormed
+            | Operation::BoxValue
+            | Operation::UnboxValue
+            | Operation::EmptyEventStore
+            | Operation::ExtendEventStore
+            | Operation::EventStoreIncludes
+            | Operation::EventStoreIncludedIn
+            | Operation::NoOp => self.internal_error(
+                id,
+                format!("unsupported specification construct: `{:?}`", op),
+            ),
+        }
+    }
+
+    fn gen_cast_call(&mut self, targets: Vec<TempIndex>, id: NodeId, args: &[Exp]) {
+        let ty = self.env().get_node_type(id);
+        let bytecode_op = match ty {
+            Type::Primitive(PrimitiveType::U8) => BytecodeOperation::CastU8,
+            Type::Primitive(PrimitiveType::U16) => BytecodeOperation::CastU16,
+            Type::Primitive(PrimitiveType::U32) => BytecodeOperation::CastU32,
+            Type::Primitive(PrimitiveType::U64) => BytecodeOperation::CastU64,
+            Type::Primitive(PrimitiveType::U128) => BytecodeOperation::CastU128,
+            Type::Primitive(PrimitiveType::U256) => BytecodeOperation::CastU256,
+            _ => {
+                self.internal_error(id, "inconsistent type");
```

Comment:
Ok, so I assume that cast calls only work for uint types.  But how should I know that?  Is there a doc somewhere I should refer to?

Reply:
That would be the Move book (at least I hope it clarifies this), which we host at aptos.dev.

Reply:
Indeed.  "Integer types of one size can be cast to integer types of another size. Integers are the only types in Move that support casting."  (https://move-language.github.io/move/integers.html).

I don't think it would be terrible to add that as a comment: "Integers are the only types in Move that support casting."

Reply:
Sorry not sure why I should add basic documentation about the language -- which is already encoded in the bytecode format as it does not support any other cast operation -- here.

Reply:
Fine.  It wouldn't be terrible, but it's not necessary.